openjdk version confusion in checkJava

When I try to `make` I get a complaint that I don't have a `java` of the required version:

```
~/src/k/latest/evm-semantics(dapphub/stable) » make build-java
== kompile: .build/java/driver-kompiled/timestamp
/home/lev/src/k/latest/evm-semantics/.build/k/k-distribution/target/release/k/bin/kompile --debug --main-module ETHEREUM-SIMULATION --backend java \
				--syntax-module ETHEREUM-SIMULATION .build/java/driver.k --directory .build/java -I .build/java
Error: K requires Java 1.8 to run but the detected version is 10.0.1.
Please either add Java 1.8 bin directory to the PATH or set the JAVA_HOME
environment variable accordingly.
Makefile:131: recipe for target '.build/java/driver-kompiled/timestamp' failed
make: *** [.build/java/driver-kompiled/timestamp] Error 2
```
Indeed, the version reported by `$ java -version` is:
```
~ » java -version
openjdk version "10.0.1" 2018-04-17
OpenJDK Runtime Environment (build 10.0.1+10-Debian-4)
OpenJDK 64-Bit Server VM (build 10.0.1+10-Debian-4, mixed mode)
```
So this appears to be some confusion in the Java version number convention. By the way, I am using `openjdk-10-jre 10.0.1+10-4` provided by debian testing. I can only assume that anyone else using debian would have the same problem. The problem is resolved by adjusting `MIN_VERSION` in `k-distribution/src/main/scripts/lib/checkJava`, after that everything builds as expected.

@ehildenb

--- a/k-distribution/src/main/scripts/lib/checkJava
+++ b/k-distribution/src/main/scripts/lib/checkJava
@@ -47,7 +47,7 @@ fi
 if [[ "$_java" ]]; then
     version=$(echo $version | awk -F '"' '/version/ {print $2}')
 #    echo version "$version"
-    if ! [[ "$version" > "$MIN_VERSION" ]]; then
+    if ! [ "$MIN_VERSION" = "`printf "$MIN_VERSION\n%s" "$version" | sort -V | head -1`" ]; then
         echoerr -e "\033[1;31mError:\033[0m K requires Java $MIN_VERSION to run but the detected version is $version."
         echoerr "Please either add Java $MIN_VERSION bin directory to the PATH or set the JAVA_HOME
 environment variable accordingly."

--- a/pom.xml
+++ b/pom.xml
@@ -237,7 +237,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-enforcer-plugin</artifactId>
-        <version>1.4.1</version>
+        <version>3.0.0-M2</version>
         <configuration>
           <rules>
             <requireJavaVersion>

